src/vscode/impl/recommendationServiceImpl.ts

```diff
@@ -40,6 +42,18 @@ export class RecommendationServiceImpl implements IRecommendationService {
         return path.resolve(context.globalStorageUri.fsPath, '..', 'vscode-extension-recommender');
     }
 
+    protected getExtensionConfiguration(): string | undefined {
+        if (workspace.workspaceFolders !== undefined) {
+            for (const wf of workspace.workspaceFolders) {
```

Comment:
According to https://code.visualstudio.com/docs/editor/extension-marketplace#_workspace-recommended-extensions, in a multi-root workspace, we should check `.code-workspace`.